src/lwc/psElevateTokenHandler/psElevateTokenHandler.js

```diff
@@ -0,0 +1,168 @@
+    */
+    getVisualforceOriginURLs(domainInfo) {
+        const namespace = this.getCurrentNamespace();
+
+        let url = `https://${domainInfo.orgDomain}--c.visualforce.com`;
+        let alternateUrl = `https://${domainInfo.orgDomain}--c.${domainInfo.podName}.visual.force.com`;
+
+        if (namespace) {
+            url = url.replace('--c', `--${namespace}`);
+            alternateUrl = alternateUrl.replace('--c', `--${namespace}`);
+        }
+
+        return [{ value: url }, { value: alternateUrl }];
+    }
+
+    /***
+    * @description Returns the NPSP namespace (if any)
+    */
+    getCurrentNamespace() {
+        return getNamespace(PAYMENT_AUTHORIZATION_TOKEN_FIELD.fieldApiName);
+    }
+
+    /***
+    * @description Dispatches the application event when the Elevate credit card iframe 
+    * is displayed or hidden 
+    */
+    dispatchApplicationEvent(eventName, payload) {
+        fireEvent(null, eventName, payload);
+    }
+
+    /***
+    * @description Listens for a message from the Visualforce iframe.
+    * Rejects any messages from an unknown origin.
+    */
+    registerPostMessageListener(component) {
+        window.onmessage = async function (event) {
+            component.visualforceOrigin = component.visualforceOriginUrls.find(
+                origin => event.origin === origin.value
+            ).value;
+
+            if (component.visualforceOrigin) {
+                const message = JSON.parse(event.data);
+                component.handleMessage(message);
+            }
+        }
+    }
+
+    /***
+    * @description Handles messages received from Visualforce page.
+    * @param message Message received from the iframe
+    */
+    handleMessage(message) {
+        if (message.error || message.token) {
+            if (isFunction(this.tokenCallback)) {
+                this.tokenCallback(message);
+            }
+        }
+    }
+
+    /***
+    * @description Method sends a message to the visualforce page iframe requesting a token. 
+    * This request response is found and handled in the registerPostMessageListener().
+    * @param visualforceOrigin Visualforce origin
+    * @param iframe The payment services iframe displayed within the credit card widget LWC
+    * @param handleError An error handler function
+    * @return Promise A token promise
+    */
+    requestToken(visualforceOrigin, iframe, handleError) {
+        if (isNull(iframe)) {
+            return;
+        }
+
+        const tokenPromise = new Promise((resolve, reject) => {
+
+            const timer = setTimeout(() =>
+                reject(this.handleTimeout(handleError)),
+                TOKENIZE_TIMEOUT_MS
+            );
+
+            this.tokenCallback = message => {
+                clearTimeout(timer);
+
+                if (message.error) {
+                    reject(handleError(message));
+
+                } else if (message.token) {
+                    resolve(message.token);
```

Comment:
@ptewson-sfdo , the resolve contains the token, no field setting in the new common token handler class.